magefile.go

```diff
@@ -1154,15 +1216,319 @@ func packageAgent(platforms []string, packagingFn func()) {
+	}
+
+	c := new(http.Client)
+	resp, err := c.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching latest build using %q: %w", branchLatestBuildUrl, err)
+	}
+	if mg.Verbose() {
+		log.Printf("Received response for %q : %+v", branchLatestBuildUrl, resp)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
+		return nil, fmt.Errorf("bad HTTP status for GET %q: %d - %q", branchLatestBuildUrl, resp.StatusCode, resp.Status)
+	}
+
+	bi := new(branchInfo)
+	// consume body
+	err = json.NewDecoder(resp.Body).Decode(bi)
+	if err != nil {
+		return nil, fmt.Errorf("decoding json branch information: %w", err)
+	}
+
+	if mg.Verbose() {
+		log.Printf("Received branch information for %q: %+v", branch, bi)
+	}
+
+	// return branch info
+	return bi, nil
+}
+
+func mapManifestPlatformToAgentPlatform(manifestPltf string) (string, bool) {
+	mappings := map[string]string{
+		"linux-x86_64":   "linux-amd64",
+		"windows-x86_64": "windows-amd64",
+		"darwin-x86_64":  "darwin-amd64",
+		"darwin-aarch64": "darwin-arm64",
+		"linux/x86_64":   "linux/amd64",
+		"windows/x86_64": "windows/amd64",
+		"darwin/x86_64":  "darwin/amd64",
+		"darwin/aarch64": "darwin/arm64",
+	}
+
+	mappedPltf, found := mappings[manifestPltf]
+	if !found {
+		// default to the manifest platform if no mapping is found
+		mappedPltf = manifestPltf
+	}
+
+	return mappedPltf, found
+
+}
+
+func filterPackagesByPlatform(pkgs map[string]tools.Package) map[string]tools.Package {
+	if mg.Verbose() {
+		log.Printf("unfiltered packages: %v", pkgs)
+	}
+	platforms := devtools.Platforms.Names()
+	filteredPackages := map[string]tools.Package{}
+	for pkgName, pkgDesc := range pkgs {
+		if mg.Verbose() {
+			log.Printf("checking if %s:%v should be included", pkgName, pkgDesc)
+		}
+		for _, pkgOS := range pkgDesc.Os {
+			platformString, _ := mapManifestPlatformToAgentPlatform(fmt.Sprintf("%s/%s", pkgOS, pkgDesc.Architecture))
+			if slices.Contains(platforms, platformString) {
+				if mg.Verbose() {
+					log.Printf("platforms include %s", platformString)
+				}
+				filteredPackages[pkgName] = pkgDesc
+				break
+			}
+		}
+	}
+	if mg.Verbose() {
+		log.Printf("filtered packages: %v", filteredPackages)
+	}
+	return filteredPackages
+}
+
+func downloadDRAArtifacts(ctx context.Context, manifestUrl string, downloadDir string, projects ...string) (map[string]tools.Package, error) {
+
+	build, err := manifest.DownloadManifest(manifestUrl)
+	if err != nil {
+		return nil, fmt.Errorf("downloading manifest from %q: %w", manifestUrl, err)
+	}
+
+	// Create a dir with the buildID at <downloadDir>/<buildID>
+	draDownloadDir := filepath.Join(downloadDir, build.BuildID)
+	err = os.MkdirAll(draDownloadDir, 0o770)
+
+	if err != nil {
+		return nil, fmt.Errorf("creating %q directory: %w", err)
+	}
+
+	if mg.Verbose() {
+		log.Printf("url:%q, err %v", manifestUrl, err)
```

Comment:
[Blocker]
`err` is `nil`, why are you printing it here?
Also, just printing `url` will likely be confusing, better to write it in full
```suggestion
		log.Printf("manifest url:%q", manifestUrl)
```

Reply:
Side effect of refactoring, will remove